src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,42 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.UserNotAllowed;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_LOGIN_LENGTH = 6;
+    private static final int MINIMAL_USER_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
+    public User register(User user) throws UserNotAllowed {
+        if (user != null) {
+            for (User item : Storage.people) {
+                if (item.getLogin().equals(user.getLogin())) {
+                    throw new UserNotAllowed("User with login "
+                            + user.getLogin() + " alredy exist");
+                }
```

Comment:
Iterating over `Storage.people` to check for duplicate logins is not efficient. Consider using a Set or Map to store users by their login to achieve O(1) lookup time.